manifests/init.pp

```diff
@@ -188,6 +191,7 @@
   Stdlib::Fqdn $servername = $facts['networking']['fqdn'],
   Array[Stdlib::Absolutepath] $allowed_import_path = ['/var/lib/pulp/sync_imports'],
   Array[Stdlib::Absolutepath] $allowed_export_path = [],
+  Array[String] $allowed_content_checksums = ['sha224', 'sha256', 'sha384', 'sha512'],
```

Comment:
Should this be Array[Enum[…]] to limit possibilities to only known ones? Or would that be to limiting for future pulpcore releases?

Reply:
I think that makes sense?  we'd need to update it anyways to adjust the default i would think?  Its unlikely to be a frequent occurrence either :)

Reply:
added!

Reply:
If pulpcore releases a new supported checksum, instead of say puppet-foreman_proxy_content updating what it passes to this parameter to include that new parameter, a release of this module will have to be made to update the list of allowed values to then allow the update to puppet-foreman_proxy_content as a consumer. It is balancing specifying only "valid" values and future proofing. Does Pulp throw an error if an incorrect value is supplied?

Reply:
You get an error on startup of the workers:
django.core.exceptions.ImproperlyConfigured: ALLOWED_CONTENT_CHECKSUMS may only contain algorithms known to pulp - see constants.ALL_KNOWN_CONTENT_CHECKSUMS for the allowed list.

Also, sha256 is required, it won't startup without, not sure if we should handle that here.  @ehelms you bring up a good point, maybe we should drop the Enum here and only have it in the foreman_proxy_content change?

Reply:
I lean towards not having the Enum personally to allow easier future proofing. This is usually where @ekohl shows up and sets us all straight :)

Reply:
If you want to reuse it, you can define it a type alias. https://github.com/theforeman/puppet-foreman_proxy/blob/master/types/listenon.pp is probably the easiest example from our codebase. So you would end up with
```puppet
type Pulpcore::ContentChecksums = Array[Enum[...], 1]
```

Then you can use, both in this module and puppet-foreman_proxy_content, `Pulpcore::ContentChecksums $allowed_content_checksums`. Benefits are that you only need to maintain it in one place while still having strong validation.

https://rspec-puppet.com/documentation/type_aliases/ documents how to write simple tests for it.

It's not that often that a new checksum type is added and it needs actual code on the Pulp side. Given that there is a list of constants, I do lean to an enum. Especially if we plan to expose this to users. Debugging a failed service start is a much worse experience than an up front error.